reactioncommerce/reaction
# refactor: source SMTP email configuation from ENV vars instead of DB

Resolves #5759 
Impact: minor
Type: refactor

## Issue
Settings for SMTP email service were being saved to the DB in plain text which is not secure. This PR refactors sourcing SMTP email configuration from the `MAIL_URL` env var, this will also help us move closer to PCI compliance.

## Solution
Source SMTP email config from ENV var.

If you're solving a UIX related issue, please attach screen-caps or gifs showing how your solution differs from the issue.

**NOTE:**
I have created a new issue to remove the old UI used to set these settings in the Admin UI. https://github.com/reactioncommerce/reaction-admin/issues/84
## Breaking changes
NA

## Testing
1. Set the `MAIL_URL` to a desired value, i.e. `MAIL_URL=https://user:pass@example.com:465`
2. Restart the API
2. Use the `verifySMTPEmailSettings` to verify the settings are correct.
```
mutation verifySMTPEmailSettings($input: VerifySMTPEmailSettingsInput!) {
  verifySMTPEmailSettings(input: $input) {
    isVerified
  }
}
```
variables:
```
{
  "input": {
    "shopId": "cmVhY3Rpb24vc2hvcDpKOEJocTN1VHRkZ3daeDNyeg=="
  }
}
```
